src/MapViewController.swift

```diff
@@ -813,6 +893,40 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
     tgViewController.mapViewDelegate = self
   }
 
+  override open func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    //We only want to attempt to reload incase we reloaded due to memory issues
+    if (stateSaver != nil) {
+      print("reloading tgviewcontroller due to memory warning removal")
+      reloadTGViewController()
+    }
+  }
+
+  override open func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    isCurrentlyVisible = true
+  }
+
+  override open func viewDidDisappear(_ animated: Bool) {
+    super.viewDidDisappear(animated)
+    isCurrentlyVisible = false
+  }
+
+  override open func didReceiveMemoryWarning() {
+    if (!isCurrentlyVisible) {
+      tgViewController.willMove(toParentViewController: nil)
+      tgViewController.view.removeConstraints(tgViewController.view.constraints)
+      tgViewController.view.removeFromSuperview()
+      tgViewController.removeFromParentViewController()
+      stateSaver = StateReclaimer(tilt: tilt, rotation: rotation, zoom: zoom, position: position, cameraType: cameraType, mapStyle: currentStyle)
+      //Probably unnecessary, but just incase we don't want any calls coming through from older versions that the OS has yet to clean up
+      tgViewController.gestureDelegate = nil
+      tgViewController.mapViewDelegate = nil
```

Comment:
Think I have this working now. I think this broke as part of the refactor to TGMarker but I could be wrong.